config/initializers/monkeypatches/departure.rb

```diff
@@ -1,11 +1,13 @@
+# Fix for https://github.com/departurerb/departure/issues/110, where advisory locks are not released after migration
+# is completed. This should be removed when a fix is available upstream.
 module Departure
-  module Migration
-    # Make all connections in the connection pool NOT use PerconaAdapter instead of the current adapter.
-    # Ref https://github.com/departurerb/departure/issues/110.
-    def reconnect_with_percona
-      ActiveRecord::Base.establish_connection(connection_config.merge(adapter: "percona"))
+  class ConnectionBase < ActiveRecord::Base # rubocop:disable Rails/ApplicationRecord
+    def self.establish_connection(config = nil)
+      super
     end
   end
+
+  class OriginalAdapterConnection < ConnectionBase; end
```

Comment:
Is this still needed?

Reply:
Never mind, I see your commit message now. This is to fully reverse the patch at https://github.com/departurerb/departure/blob/982966c8e57a1cb0c3a1c0083cc3d3d0b74da51d/lib/departure/connection_base.rb.